examples/camera-app/linux/src/uploader/pushav-uploader.cpp

```
@@ -0,0 +1,122 @@
+/*
+ *
+ *    Copyright (c) 2025 Project CHIP Authors
+ *    All rights reserved.
+ *
+ *    Licensed under the Apache License, Version 2.0 (the "License");
+ *    you may not use this file except in compliance with the License.
+ *    You may obtain a copy of the License at
+ *
+ *        http://www.apache.org/licenses/LICENSE-2.0
+ *
+ *    Unless required by applicable law or agreed to in writing, software
+ *    distributed under the License is distributed on an "AS IS" BASIS,
+ *    WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ *    See the License for the specific language governing permissions and
+ *    limitations under the License.
+ */
+
+#include "pushav-uploader.h"
+#include <lib/support/logging/CHIPLogging.h>
+
+PushAVUploader::PushAVUploader() : running(false) {}
+
+PushAVUploader::~PushAVUploader()
+{
+    stop();
+}
+
+void PushAVUploader::process_queue()
+{
+    while (running)
+    {
+        std::pair<std::string, std::string> data;
+        {
+            std::lock_guard<std::mutex> lock(queue_mutex);
+            if (!av_data.empty())
+            {
+                data = av_data.front();
+                av_data.pop();
+            }
+        }
+        if (!av_data.empty())
```

After popping, the test if (!av_data.empty()) decides whether to call upload_data. The queue may be empty after you popped the last element, so the item you just removed is silently dropped.

---

Done